tools/onenote-migration/onenote_migrate.py

```
@@ -0,0 +1,567 @@
+    Parse Git log for commit messages that embed a page ID.
+    Commit messages use the format: 'page:<PAGE_ID> <title>'
+    Returns a set of already-migrated page IDs.
+    """
+    migrated: set[str] = None
+    log_path = repo_path / ".git"
+    if not log_path.exists():
+        return set()
+
+    try:
+        import subprocess
+        result = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)